third_party/dml/DirectMLX.h

```diff
@@ -247,79 +248,132 @@ namespace dml
     class Scope;
     class Expression;
 
-    enum class TensorLayout
+    using TensorDimensions = SmallVector<uint32_t, 4>;
+
+    // The custom properties returned by a TensorPolicy.
+    struct TensorProperties
     {
-        Default = 0,
-        Nchw = 0,
-        Nhwc = 1,
+        Optional<TensorDimensions> strides;
+        uint64_t totalTensorSizeInBytes;
+        uint32_t guaranteedBaseOffsetAlignment;
     };
```

Comment:
Now that we have a strides policy, do we still need to have a "strides" parameter as part of the gather operators?

Reply:
Nope, which is why they've been removed. The int64 striding is now handled by a custom tensor policy (see dml_util.cc)